src/reducers/versions.tsx

```diff
@@ -107,6 +107,15 @@ type PartialExternalVersion = {
   version: string;
 };
 
+export const changeTypes: Record<string, ChangeType> = Object.freeze({
```

Comment:
This is an interesting approach, and I like the way it has simplified the other files that were having to cast strings as `ChangeType`, but I'm a bit uncomfortable with it because what we are creating here is not a Type, but rather an object that contains Types. This isn't something we've ever done before, and I'm concerned that it might be confusing to someone else encountering the code. 

Is there a way to make this object into an actual Type? I guess that's basically what an `enum` is. One thought I had before looking at the code was this: If the problem is that we cannot define an `enum` in a `.d.ts` file, could we define the `enum` in here and then import it into the `.d.ts` file?

@willdurand do you have an opinion on whether you think this approach (creating an object that contains Types) is one that would be good to implement?

Reply:
That sounds good to me. It is not really different than what we're doing with constants. I am fine with having an object although maybe we could use plain old constants instead:

```js
const DELETE: 'delete' = 'delete';
...
```

Reply:
@willdurand Do you mean in the `index.d.ts` file, like:

```js

  const DELETE = 'delete';
  const INSERT = 'insert';
  ...

  type ChangeType =
    | DELETE
    | INSERT
    ...;
```

Reply:
No, we shouldn't touch the definition file. The definition file is a TypeScript "hack", which does not get compiled to JS (which makes TS not a language but a hack too btw...). Anyway, the constants would be do avoid duplication (but with the right TS type at least).

I guess having a map would work too (as it is currently in this patch) because we would only have to import the map and not each constant (but the result would be the same).